core/store/src/trie/shard_tries.rs

```diff
@@ -442,22 +471,69 @@ impl KeyForStateChanges {
         key
     }
 
+    /// Without changing the existing TrieKey format, encodes ShardUId into the row key.
+    /// See `delayed_receipt_key_from_trie_key` for decoding this row key.
+    pub fn delayed_receipt_key_from_trie_key(
+        block_hash: &CryptoHash,
+        trie_key: &TrieKey,
+        shard_uid: &ShardUId,
+    ) -> Self {
+        let mut key = Self::new(block_hash, trie_key.len());
```

Comment:
nit: the reserved capacity should be trie_key.len() + size of shardUid, right? (as we also add the version and shard_id

Reply:
Done